webfront/tests/tests_structure_endpoint_protein_filter.py

```diff
@@ -197,40 +195,40 @@ def test_urls_that_should_fail_with_no_content(self):
         prot_s = "M5ADK6"
         prot_t = "P16582"
         tests = [
-            "/api/structure/pdb/" + pdb + "/protein/uniprot/bad_uniprot",
-            "/api/structure/pdb/" + pdb + "/protein/unreviewed/" + prot_s,
-            "/api/structure/pdb/protein/unreviewed/" + prot_s,
-            "/api/structure/pdb/protein/reviewed/" + prot_t,
+            f"/api/structure/pdb/{pdb}/protein/uniprot/bad_uniprot",
+            f"/api/structure/pdb/{pdb}/protein/unreviewed/{prot_s}",
+            f"/api/structure/pdb/protein/unreviewed/{prot_s}",
+            f"/api/structure/pdb/protein/reviewed/{prot_t}",
         ]
         for url in tests:
             self._check_HTTP_response_code(
-                url, msg="The URL [" + url + "] should've failed."
+                url, msg=f"The URL [{url}] should've failed."
             )
 
     def test_urls_that_should_fail(self):
         pdb = "1JM7"
         prot_s = "M5ADK6"
         tests = [
-            "/api/structure/pdb/bad_structure/protein/uniprot/" + prot_s,
-            "/api/bad_endpoint/pdb/" + pdb + "/protein/uniprot/" + prot_s,
-            "/api/structure/bad_db/" + pdb + "/protein/uniprot/" + prot_s,
-            "/api/structure/pdb/" + pdb + "/protein/uniprot/bad_protein",
-            "/api/structure/pdb/" + pdb + "/protein/bad_db/" + prot_s,
-            "/api/structure/pdb/" + pdb + "/bad_endpoint/uniprot/" + prot_s,
+            f"/api/structure/pdb/bad_structure/protein/uniprot/{prot_s}",
+            f"/api/bad_endpoint/pdb/{pdb}/protein/uniprot/{prot_s}",
+            f"/api/structure/bad_db/{pdb}/protein/uniprot/{prot_s}",
+            f"/api/structure/pdb/{pdb}/protein/uniprot/bad_protein",
+            f"/api/structure/pdb/{pdb}/protein/bad_db/{prot_s}",
+            f"/api/structure/pdb/{pdb}/bad_endpoint/uniprot/{prot_s}",
         ]
         for url in tests:
             self._check_HTTP_response_code(
                 url,
                 code=status.HTTP_404_NOT_FOUND,
-                msg="The URL [" + url + "] should've failed.",
+                msg=f"The URL [{url}] should've failed.",
             )
         tests = [
-            "/api/structure/pdb/BADP/protein/uniprot/" + prot_s,
-            "/api/structure/pdb/" + pdb + "/protein/uniprot/bad_prot",
+            f"/api/structure/pdb/BADP/protein/uniprot/{prot_s}",
```

Comment:
Mhh... dunno. So we get the `404` when the request is malformed, like using the wrong endpoint, or an accession that doesn't match the regex. but is the URL Is OK we are considering as an empty queryset returning `204`. We doing this even at single endpoint level. For example:
* https://www.ebi.ac.uk/interpro/api/structure/PDB/badp => `204`
* https://www.ebi.ac.uk/interpro/api/structure/PDB/randomACC => `404`

I guess is open to discussion what is the best approach for it, but just mentioning this is an API wide strategy, so a change for it will affect every endpoint.